Added hgn_dreadnaught.ship and meg_dreadnaughtberth.ship

SaintZFX opened the pull request:
meg_dreadnaughtberth.ship is vanilla apart from:
NewShipType.maxhealth=400000000
NewShipType.thrusterMaxSpeed=0
NewShipType.mainEngineMaxSpeed=0
This stops it from being accidentally destroyed if placed as a set piece in a map.

hgn_dreadnaught made capturable by changing/adding:
setSupplyValue(NewShipType, "Battlecruiser", 1.0)
addAbility(NewShipType,"CanBeSalvageCaptured",0,1,0,0,1,"SalCap")
Changed HP to be in line with Battlecruisers.

SaintZFX committed:
Added hgn_dreadnaught.ship and meg_dreadnaughtberth.ship

meg_dreadnaughtberth.ship is vanilla apart from:
NewShipType.maxhealth=400000000
NewShipType.thrusterMaxSpeed=0
NewShipType.mainEngineMaxSpeed=0
This stops it from being accidentally destroyed if placed as a set piece in a map.

hgn_dreadnaught made capturable by changing/adding:
setSupplyValue(NewShipType, "Battlecruiser", 1.0)
addAbility(NewShipType,"CanBeSalvageCaptured",0,1,0,0,1,"SalCap");

Novaras commented:
instead of adding dreadnaughts to the bc supply count, instead we should try to add it to some other supply count

perhaps a 'misc' one for anything we want to place on maps which could be capturable

there is already a `supplyLimit("SinglePlayerMisc", 100);`, we could add `supplyLimit("MultiPlayerMisc", 100);`
then give the dreadnaught 1 supply for this new type

also, why reduce HP? this is a map gimmick unit right?

SaintZFX commented:
Vanilla value is Dreadnaught for supply limit '_setSupplyValue(NewShipType, "Dreadnaught", 1.0)_'.
It is a map gimmick, but my intention was to have it be a capturable replacement for a BC, hence I added it to Battlecruiser limit and tweaked the HP. In any case I cannot imagine 500K HP to be viable in any sort of gameplay scenario.

Novaras commented:
You can have many supply values per ship
- in `familylist.lua` the first table here is `unitcapsFamily`, you can add new supply types here (capitalisation doesn't matter)
- then in `scripts/rules/deathmatch/unitcaps` add supply limits (could initially just test with `normal.lua`)
- give the dreadnaught `1` or maybe more of this new supply type, we will reuse it for our special ships etc.
- i'd say just clone the dreanaught, rename it `mpp_dreadnaught` and alter that guy instead - you will need the `.hod`, `.events` and etc files, not just the ship file (ive attached these in case you needed them)
- same for the berth

thanks for helping, is this in prep for a new map?

[hgn_dreadnaught.zip](https://github.com/Novaras/mp-patch/files/6910607/hgn_dreadnaught.zip)
[meg_dreadnaughtberth.zip](https://github.com/Novaras/mp-patch/files/6910614/meg_dreadnaughtberth.zip)

SaintZFX committed:
Added mpp_dreadnaught and modified unit caps to include Dreadnaught

Removed hgn_dreadnaught.ship
Added mpp_dreadnaught (copy of hgn_dreadnaught) and made it caputrable. Modified unit caps to add Dreadnaught.
